chain/network/src/testonly/actix.rs

```diff
@@ -12,6 +12,9 @@ impl Thread {
 
 impl Drop for Thread {
     fn drop(&mut self) {
+        if std::thread::panicking() {
+            return;
+        }
         self.0.take().unwrap().join().unwrap().unwrap();
```

Comment:
```suggestion
        let h = self.0.take().unwrap().join();
        if !std::thread::panicking() {
            h.unwrap().unwrap();
        }
```

Joining even with panic would be marginally better I think

Reply:
moved to a separate PR